### PR TITLE
feat: use component-compiler-utils

### PR DESCRIPTION
**Pull request by znck**
There is simple `assemble` implementation which can cater most use-cases.

### CODE REVIEW

**Comment by yyx990803** [comment]
I assume this is WIP until tests are passing?

**Comment by znck** [comment]
Yes. This assumes component-compiler-utils PRs are merged. I asked for your review on approach. The  descriptor compiler and an assemble implementation to cater common use cases.

File: src/compiler.ts [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    const scopeId =
+      'data-v-' +
+      (this.template.isProduction
+        ? hash(path.basename(filename) + source)
+        : hash(filename + source))
+
+    const template =
+      descriptor.template && this.compileTemplate(filename, descriptor.template)
+
+    const styles = descriptor.styles.map(style => {
```
Comment (yyx990803) [inline src/compiler.ts]: The map function should probably be extracted into a method.

**Comment by yyx990803** [comment]
The externalization can be introduced later. Once you finish the style map refactor you can merge this to master and publish as `@vue/component-compiler`. Also add some documentation :)

**Pull request merged**